### DISCUSSION TITLE
Creating a --release build generates a _Test AppPackage.

### DISCUSSION THREAD

**Discussion by warrenronsiek** [microsoft/react-native-windows#13333] [Q&A] [answered]
When I run `npx react-native run-windows --release --no-launch --no-deploy` in a clean windows CCI instance, The artifacts generated in the AppPackage directory have a `_Test` suffix. E.g.:
```
>> ls
MyApp_0.1.7.0_x64_Test\
MyApp_0.1.7.0_x64.msixupload
```
It seems suspicious to generate `_Test` artifacts when `--release` is specified. Is this the intended behavior?

**Comment by warrenronsiek** [comment]
After some snooping around in the cli internals and switching to verbose logging, it is generating the msbuild command with the correct Release perameter when passing `--release` so that seems good. 

What I noticed is that while the `_Test` directory seems to include install scripts and dependencies like Microsoft.Ui.Xaml, the same isn't the case for the .msixupload file archive, which if you try to install using it's contents, will give a dependency error. I'm assuming that the Microsoft store handles these dependencies and they don't actually need to be included in the application?

I'm asking because my application seems to fail to launch as part of Microsoft store certification, which uses the .msixupload, but when I "Test" things using the artifacts in the "_Test" directory to install, everything works fine.

**Comment by warrenronsiek** [comment]
Turns out the problem was unrelated to the `_Test` directory.

**Reply by chrisglein** [reply, marked as answer]
Yes, the fact that the Release build has a "_Test" suffix is expected for VS projects of this type.

**Reply by chrisglein** [reply]
Does this mean you figured out your other problem? Which I assume was this comment:

> I'm asking because my application seems to fail to launch as part of Microsoft store certification, which uses the .msixupload, but when I "Test" things using the artifacts in the "_Test" directory to install, everything works fine.

**Reply by warrenronsiek** [reply]
> Does this mean you figured out your other problem? Which I assume was this comment:
> 
> > I'm asking because my application seems to fail to launch as part of Microsoft store certification, which uses the .msixupload, but when I "Test" things using the artifacts in the "_Test" directory to install, everything works fine.

Yup. The problem had to do with FullTrustProcess not having bundled dependencies and thus failing when run on a system that didn't have .NET pre-installed. Nothing wrong with react-native-windows builds.
